+nosnoc/+discrete_time_problem/Cls.m

```diff
@@ -0,0 +1,1600 @@
+                                      case 'Abs'
+                                        if jj ~= 1 || ~opts.no_initial_impacts
+                                            n_vt_lb = obj.w.N_vt(ii,jj);
+                                        else
+                                            n_vt_lb = 0;
+                                        end
+                                        sum_n_vt = n_vt_lb + sum2(obj.w.n_vt(ii,jj,:));
+                                        if jj ~= 1 || ~opts.no_initial_impacts
+                                            Gij = vertcat(Gij, {obj.w.P_vt(ii,jj)});
+                                            Hij = vertcat(Hij, {sum_n_vt});
+                                        end
+                                        for kk=1:opts.n_s
+                                            p_vt_ijk = obj.w.n_vt(ii,jj,kk);
+                                            
+                                            Gij = vertcat(Gij, {p_vt_ijk});
+                                            Hij = vertcat(Hij, {n_vt_lb + sum_n_vt});
+                                        end
+                                      case 'Lp'
+                                        if jj ~= 1 || ~opts.no_initial_impacts
+                                            p_vt_lb = obj.w.P_vt(ii,jj);
+                                        else
+                                            p_vt_lb = 0;
+                                        end
+                                        sum_p_vt = sum2(obj.w.p_vt(ii,jj,:));
+                                        if jj ~= 1 || ~opts.no_initial_impacts
+                                            n_vt_lb = obj.w.N_vt(ii,jj);
+                                        else
+                                            n_vt_lb = 0;
+                                        end
+                                        sum_p_vt = p_vt_lb + sum2(obj.w.p_vt(ii,jj,:));
+                                        sum_n_vt = n_vt_lb + sum2(obj.w.n_vt(ii,jj,:));
+                                        for kk=1:opts.n_s
+                                            alpha_vt_ijk = obj.w.alpha_vt(ii,jj,kk);
+                                            
+                                            Gij = vertcat(Gij, {alpha_vt_ijk});
+                                            Hij = vertcat(Hij, {sum_p_vt});
+                                            Gij = vertcat(Gij, {1-alpha_vt_ijk});
+                                            Hij = vertcat(Hij, {sum_n_vt});
+                                        end
+                                    end
+                                end
+                            end
+                            obj.G.cross_comp(ii,jj) = {vertcat(Gij{:})};
+                            obj.H.cross_comp(ii,jj) = {vertcat(Hij{:})};
+                        end
+                    end
+                  case CrossCompMode.FE_FE
```

Comment:
the good news is. all cross comp modes work, the disc manipulation example converges super fast with any of the options. :)